### ISSUE TITLE
Error encountered when running the example Chinese clip with Metal

### ISSUE THREAD

**Issue by BB-fat** [huggingface/candle#2611]
`cargo run --example chinese_clip --features metal`

```
Error: Metal contiguous index_select U32 U32 not implemented
   0: std::backtrace_rs::backtrace::libunwind::trace
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:116:5
   1: std::backtrace_rs::backtrace::trace_unsynchronized
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2: std::backtrace::Backtrace::create
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/backtrace.rs:331:13
   3: candle_core::error::Error::bt
             at ./candle-core/src/error.rs:238:25
   4: <candle_core::metal_backend::MetalStorage as candle_core::backend::BackendStorage>::index_select
             at ./candle-core/src/error.rs:260:20
   5: candle_core::storage::Storage::index_select
```

**Comment by LaurentMazare** [comment]
Should be fixed in #2613

**Comment by BB-fat** [comment]
It works, amazing work!
